discord/ext/oauth/token.py

```diff
@@ -0,0 +1,20 @@
+from dataclasses import dataclass
+from datetime import datetime
+
+from ...utils import utcnow
+
+
+@dataclass
```

Comment:
A note here is that the library doesn't use dataclasses to represent Discord models generally. Whilst this may not be an issue, it may be wiser to follow our current standard.

We create payload types with a `typing.TypedDict` and store their definition in `discord.types.<item>` and we create a regular class instead.